packages/injective-cosmwasm/src/msg.rs

```diff
@@ -166,41 +167,69 @@ pub fn create_withdraw_msg(sender: Addr, subaccount_id: SubaccountId, amount: Co
 }
 
 pub fn create_subaccount_transfer_msg(
-    sender: Addr,
-    source_subaccount_id: SubaccountId,
-    destination_subaccount_id: SubaccountId,
-    amount: Coin,
-) -> CosmosMsg<InjectiveMsgWrapper> {
-    InjectiveMsgWrapper {
+    deps: &Deps<InjectiveQueryWrapper>,
+    source_subaccount_id: &SubaccountId,
+    destination_subaccount_id: &SubaccountId,
+    amount: &Coin,
+) -> StdResult<Vec<CosmosMsg<InjectiveMsgWrapper>>> {
+    let sender = subaccount_id_to_injective_address(source_subaccount_id, deps)?;
+    let to_address = subaccount_id_to_injective_address(destination_subaccount_id, deps)?;
+
+    let is_external_transfer = sender != to_address;
+    if is_external_transfer {
+        return create_external_transfer_msg(deps, source_subaccount_id, destination_subaccount_id, amount);
+    }
```

Comment:
what if `source_subaccount_id` is also a default one? shouldn't we check that as well?

Reply:
Well, I guess in current architecture it's not possible, because subaccount nonce for vaults starts from `1`.